qtop.py

```diff
@@ -1606,14 +1524,92 @@ def join_prints(self, *args, **kwargs):
         print "".join(joined_list[self.viewport.h_start:self.viewport.h_stop])
         return joined_list
 
+    def print_core_lines(self, core_user_map, print_char_start, print_char_stop, transposed_matrices, pattern_of_id, attrs,
+                         options1, options2):
+        signal(SIGPIPE, SIG_DFL)
+        if config['transpose_wn_matrices']:
+            tuple_ = [None, 'core_map', transpose_matrix(core_user_map, colored=True)]
+            transposed_matrices.append(tuple_)
+            return
+
+        for core_line in get_core_lines(core_user_map, print_char_start, print_char_stop, pattern_of_id, attrs):
+            try:
+                print core_line
+            except IOError:
+                try:
+                    signal(SIGPIPE, SIG_DFL)
+                    print core_line
+                    sys.stdout.close()
+                except IOError:
+                    pass
+                try:
+                    sys.stderr.close()
+                except IOError:
+                    pass
+
+    def display_wnid_lines(self, start, stop, highest_wn, wn_vert_labels, **kwargs):
+        """
+        Prints the Worker Node ID lines, after it colors them and adds separators to them.
+        highest_wn determines the number of WN ID lines needed  (1/2/3/4+?)
+        """
+        d = OrderedDict()
+        end_labels = config['workernodes_matrix'][0]['wn id lines']['end_labels']
+
+        if not NAMED_WNS:
+            node_str_width = len(str(highest_wn))  # 4 for thousands of nodes, nr of horizontal lines to be displayed
+
+            for node_nr in range(1, node_str_width + 1):
+                d[str(node_nr)] = "".join(wn_vert_labels[str(node_nr)])
+            end_labels_iter = iter(end_labels[str(node_str_width)])
+            print_wnid_lines(d, start, stop, end_labels_iter, transposed_matrices,
+                             color_func=self.color_plainly, args=('White', 'Gray_L', start > 0))
+            # start > 0 is just a test for a possible future condition
+
+        elif NAMED_WNS or options.FORCE_NAMES:  # names (e.g. fruits) instead of numbered WNs
```

Comment:
very few people will understand what "fruits" is; say arbitrary strings, instead